clippy_lints/src/ok_if_let.rs

```diff
@@ -41,18 +42,36 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for OkIfLet {
     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr<'_>) {
         if_chain! { //begin checking variables
             if let ExprKind::Match(ref op, ref body, ref source) = expr.kind; //test if expr is a match
-            if let MatchSource::IfLetDesugar { .. } = *source; //test if it is an If Let
-            if let ExprKind::MethodCall(_, _, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
+            if let MatchSource::IfLetDesugar { contains_else_clause } = *source; //test if it is an If Let
+            if let ExprKind::MethodCall(_, ok_span, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
             if let PatKind::TupleStruct(QPath::Resolved(_, ref x), ref y, _)  = body[0].pat.kind; //get operation
             if method_chain_args(op, &["ok"]).is_some(); //test to see if using ok() methoduse std::marker::Sized;
 
             then {
                 let is_result_type = match_type(cx, cx.tables.expr_ty(&result_types[0]), &paths::RESULT);
-                let some_expr_string = snippet(cx, y[0].span, "");
+                let mut applicability = Applicability::MachineApplicable;
+                let some_expr_string = snippet_with_applicability(cx, y[0].span, "", &mut applicability);
+                let trimmed_ok = snippet_with_applicability(cx, op.span.until(ok_span), "", &mut applicability);
```

Comment:
You could write
```suggestion
                let trimmed_ok = snippet_with_applicability(cx, op.span.with_lo(ok_span.lo - BytePos(1)), "", &mut applicability);
```

instead of trimming below.

Reply:
fixed via d8ac4b0195e7c4cb6667a9cc67c14c3df06b7322